flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateBackend.java

```diff
@@ -295,6 +297,7 @@ public RocksDBStateBackend(AbstractStateBackend checkpointStreamBackend, boolean
 	 * @param classLoader The class loader.
 	 */
 	private RocksDBStateBackend(RocksDBStateBackend original, Configuration config, ClassLoader classLoader) {
+		this.configuration = config;
```

Comment:
When do we use RocksDBStateBackend, seems we always use the **original** statebackend's conf first?

Reply:
Thanks @klion26 for your review. I'm not sure whether i quite get you meaning. But IMO this construct is only for creating a re-configured copy of the original state backend. So the configure passed in will be used first ? And also should keep this configure to a member field to reuse it to config the statebackend loaded from `StateBackendLoader#fromApplicationOrConfigOrDefault`.

Reply:
But the code below(start from line 302) seems to use the **original backend's** conf first, so I'm not sure you change now is on the right directory(maybe the change you've made can't solve the problem on the issue)?

Reply:
Hi, @klion26 I check the code again, only the `priorityQueueStateType` is configured from configuration, and the type from configuration will be used first see 
```
final String priorityQueueTypeString = config.getString(TIMER_SERVICE_FACTORY);

this.priorityQueueStateType = priorityQueueTypeString.length() > 0 ?
             PriorityQueueStateType.valueOf(priorityQueueTypeString.toUpperCase()) : original.priorityQueueStateType;

```
And other parameters are all passed by the constructor, so it  can work for the issue i think. but it looks strange indeed(some follow the original backend, some follow the config first), I will think how to make it better.

Reply:
How about the other configs? Do we need to do the same thing here for the other configs?

Reply:
other config are all follow this manner: check the original config whether set or undefined, if it is not defined then it will look into the config , at last fallback to the default config which is consistent with the doc of the `config()` 

> Creates a copy of this state backend that uses the values defined in the configuration for fields where that **were not yet specified** in this state backend.

But for the `priorityQueueStateType` there is no undefined type, it only has two: *rocksdb* and *heap*, if we have to follow the manner I think we just have to add an undefined value for `priorityQueueStateType`, but i am not sure do we need to do this.

What i am do now is just overriding the config loading from file with the original statebackend config to solve this problem.

Reply:
Do you think we need to check the config first, then fallback the original config, and at last fallback to the default config?

Reply:
I think we don't have to, I prefer to keep the former style, but only use the config in user code with higher priority than the default configuration loading from the file.